### PR TITLE
conditional EurekaInstanceConfigBean sidecar autoconfig

### PR DESCRIPTION
**Pull request by gadams00**
conditional EurekaInstanceConfigBean sidecar autoconfig 

wip fix for gh-3545

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@gadams00 Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-netflix&pullRequestId=3568)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-cloud/spring-cloud-netflix&pullRequestId=3568) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@gadams00 Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-netflix&pullRequestId=3568)!

File: spring-cloud-netflix-sidecar/src/main/java/org/springframework/cloud/netflix/sidecar/SidecarMarkerConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package org.springframework.cloud.netflix.sidecar;
```
Comment (spencergibb) [inline spring-cloud-netflix-sidecar/src/main/java/org/springframework/cloud/netflix/sidecar/SidecarMarkerConfiguration.java]: Needs license header. That is while checkstyle failed in the CI build.

Comment (gadams00) [reply spring-cloud-netflix-sidecar/src/main/java/org/springframework/cloud/netflix/sidecar/SidecarMarkerConfiguration.java]: added license header.

**Comment by gadams00** [comment]
I'm curious. Several unit tests fail in intellij locally (not via maven) SidecarApplicationTests.[ServerContextPathStatusAndHealthCheckUrls, AcceptAllSslCertificatesContext, ManagementContextPathStatusAndHealthCheckUrls]. I don't see those in circleci logs or in mvn test reports. Is that because the classes don't match the test pattern, so maven isn't running them?

**Comment by ryanjbaxter** [comment]
How are you running the tests?  We usually just run `./scripts/build.sh`.

**Comment by gadams00** [comment]
> How are you running the tests? We usually just run `./scripts/build.sh`.
I was using intellij's test runner in the IDE, i.e. right-click spring-cloud-netflix-sidecar/src/test/java, Run All Tests. It's not using maven, so the maven-surefire-plugin include patterns from spring-cloud-build aren't applied. It's not important, but it's interesting that there are tests in here that are effectively ignored.

**Comment by ryanjbaxter** [comment]
@gadams00 can you create a separate issue about the tests?

**Comment by gadams00** [comment]
> @gadams00 can you create a separate issue about the tests?

Sorry, I went on vacation, and finally got back to this. Here's that issue about failing hidden unit tests: https://github.com/spring-cloud/spring-cloud-netflix/issues/3584

**Pull request merged**